emanate/observers.py

```diff
@@ -100,21 +105,36 @@ def _process_queue(self):
             self._handle_event(event)
 
     def _handle_event(self, event):
+        is_retry = False
         try:
-            self._process_event(event)
+            while True:
+                published = self._process_event(event, is_retry=is_retry)
+                if self.incessant and not published:
+                    log.info('Will attempt to republish event.')
+                    time.sleep(0.2) # Wait briefly before trying again.
+                    is_retry = True
+                    continue
+                return
         except Exception as exc:
             log.exception(exc)
         finally:
             self._queue.task_done()
 
-    def _process_event(self, event):
+    def _process_event(self, event, is_retry=False):
+        '''Returns true if the event was published, false otherwise.'''
         properties = Properties()
         if event.context:
             properties.headers = event.context
 
         try:
             self.publisher.publish(self.exchange, '', event.data, properties)
+            return True
         except (AMQPError, RecursionError) as exc:
```

Comment:
Depending on the type of exception, might it sometimes be necessary to delete the publisher channel prior to retrying?  Should we consider creating a common exception handling routine based on 8f54902b?